gnome-session/gsm-manager.c

```diff
@@ -756,11 +757,36 @@ on_phase_timeout (GsmManager *manager)
         return FALSE;
 }
 
+static gboolean
+_autostart_delay_timeout (GsmApp *app)
+{
+        GError *error = NULL;
+        gboolean res;
+
+        if (!gsm_app_peek_is_disabled (app)
+            && !gsm_app_peek_is_conditionally_disabled (app)) {
+                res = gsm_app_start (app, &error);
+                if (!res) {
+                        if (error != NULL) {
+                                g_warning ("Could not launch application '%s': %s",
+                                           gsm_app_peek_app_id (app),
+                                           error->message);
+                                g_error_free (error);
+                        }
+                }
+        }
+
+        g_object_unref (app);
```

Comment:
Could we add a comment here saying that we unref here because we ref `app` when this function is called? When I first saw it here, I thought it was kinda strange to do it here, and had to read the rest of the diff for it to make sense.